src/goto-symex/symex_assign.cpp

```diff
@@ -74,7 +74,7 @@ void goto_symext::symex_assign(
     if(state.source.pc->source_location.get_hide())
       assignment_type=symex_targett::assignment_typet::HIDDEN;
 
-    guardt guard{true_exprt{}}; // NOT the state guard!
+    guardt guard{true_exprt{}, guard_manager}; // NOT the state guard!
```

Comment:
It looks like it could be replaced by a conjunction, the only operation we do on them is adding more conjuncts. I can make this type to `exprt::operandst` and rename it `guard_conjuncts` if you think that's a good idea.

Reply:
That would be nice, thank you!

Reply:
And while at it, avoid use the term "guard" in this? Maybe `lhs_if_then_else_conditions`? And then of course kill that comment ("NOT the state guard"). It's not a guard, it shouldn't be named "guard", and thus the confusion is gone.

Reply:
`lhs_if_then_else_conditions` is not very clear to me, maybe because I don't really understand what this represents. It seems like it's conditions that have to hold for the assignment to take place, in that case it could be named `assignment_conditions`?

Reply:
I'm not sure about the best name either. Yes, it is a list of conditions for an assignment to use a particular symbol on the LHS. An example: `(cond ? foo : bar) = 42` will yield:
```
foo = 42  [assignment condition: cond]
bar = 42  [assignment condition: !cond]
```
(FWIW: those if-then-else on the left-hand side are a result of dereferencing)